apps/documentation/src/components/MdxProvider/MdxTitle/MdxTitle.tsx

```diff
@@ -1,15 +1,52 @@
-import type { ElementType, JSX, ReactNode } from 'react'
-import { Title, type TitleProps } from '@mantine/core'
+import {
+  useCallback,
+  type ElementType,
+  type JSX,
+  type MouseEvent,
+  type ReactNode,
+} from 'react'
+import { Title, Anchor, type TitleProps } from '@mantine/core'
+import { IconLink } from '@tabler/icons-react'
+import { useHover } from '@mantine/hooks'
 
 export default function MdxTitle(props: TitleProps): JSX.Element {
```

Comment:
Why not use destructing here?

```suggestion
export default function MdxTitle({ order, children, ...rest }: TitleProps): JSX.Element {
```